deepchem/models/torch_models/layers.py

```diff
@@ -4792,3 +4792,129 @@ def forward(self, dist: torch.Tensor) -> torch.Tensor:
+    to help generate the correct token in the sequence.
+
+    Examples
+    --------
+    >>> from deepchem.models.torch_models.layers import DecoderRNN
+    >>> import torch
+    >>> embedding_dimensions = 5
+    >>> num_output_tokens = 7
+    >>> num_input_tokens = 12
+    >>> max_length = 4
+    >>> batch_size = 2
+    >>> device = 'cuda' if torch.cuda.is_available() else 'cpu'
+    >>> layer = DecoderRNN(embedding_dimensions, num_output_tokens,
+    ...                                 max_length, device)
+    >>> embeddings = torch.randn(batch_size, num_input_tokens, embedding_dimensions)
+    >>> output, hidden, _ = layer([embeddings,
+    ...                            embeddings[:, -1].unsqueeze(0).contiguous(), None])
+    >>> output.shape
+    torch.Size([2, 4, 7])
+
+    References
+    ----------
+    .. [1] Sutskever et al., "Sequence to Sequence Learning with Neural Networks"
+
+    """
+
+    def __init__(self,
+                 hidden_size:int,
+                 output_size:int,
+                 max_length:int,
+                 device: torch.device,
+                 **kwargs):
+        """Initialize the DecoderRNN layer.
+        
+        Parameters
+        ----------
+        hidden_size: int
+            The number of features in the hidden state.
+        output_size: int
+            The number of expected features.
+        max_length: int
+            The maximum length of the sequence.
+        device: torch.device
+            The device on which the layer is initialized.
+
+        """
+        super(DecoderRNN, self).__init__(**kwargs)
+        self.embedding = nn.Embedding(output_size, hidden_size)
+        self.gru = nn.GRU(hidden_size, hidden_size, batch_first=True)
+        self.out = nn.Linear(hidden_size, output_size)
+        self.act = get_activation("log_softmax")
+        self.step_act = get_activation("relu")
+        self.MAX_LENGTH = max_length
+        self.device = device
+
+    def __repr__(self) -> str:
+        """Returns a string representing the configuration of the layer.
+
+        Returns
+        -------
+        hidden_size: int
+            Number of features in the hidden state.
+        output_size: int
+            Number of expected features.
+        max_length: int
+            Maximum length of the sequence.
+        batch_size: int
+            Batch size of the input.
+
+        """
+        return f'{self.__class__.__name__}(hidden_size={self.hidden_size}, output_size={self.output_size}, max_length={self.max_length}, batch_size={self.batch_size})'
+
+    def forward(self, inputs: List[torch.Tensor]):
+        """
+        Parameters
+        ----------
+        inputs: List[torch.Tensor]
+            A list of tensor containg encoder_output, encoder_hidden and target_tensor
+
+        Returns
+        -------
+        decoder_outputs: torch.Tensor
+            Predicted output sequences
+        decoder_hidden: torch.Tensor
+            Hidden state of the decoder
+
```

Comment:
Maybe explain why the 3rd output is `None` here

Reply:
removed, seems more unnecessary than need to explain